config/config.go

```diff
@@ -268,6 +276,25 @@ func mergeConfigWithIncludedConfig(config *TerragruntConfig, includedConfig *Ter
 	return includedConfig, nil
 }
 
+// Merge the extra arguments prioritizing those defined in the childExtraArgs
+func mergeExtraArgs(terragruntOptions *options.TerragruntOptions, childExtraArgs []TerraformExtraArguments, parentExtraArgs *[]TerraformExtraArguments) {
+	result := *parentExtraArgs
+addExtra:
+	for _, child := range childExtraArgs {
+		for i, parent := range result {
+			if parent.Name == child.Name {
+				terragruntOptions.Logger.Printf("extra_arguments '%v' from child overriding parent", child.Name)
+				// For extra args, we want to keep the values specified in the child and put them after
+				// the parent ones, so if we encounter a duplicate, we just overwrite it.
+				result[i] = child
+				continue addExtra
```

Comment:
That seems like the proper behavior. That said, I'm not a fan of two nested loops plus a label & continue (ie goto). I had to re-read the code a few times to understand it. Would you mind keeping the same behavior, but changing the code as described above: that is, two simple, un-nested loops that load elements into a list if that element isn't in the list already?